### ISSUE TITLE
UFC Undisputed 3 [BLES01231] Audio-Cuts and Crashing

### ISSUE THREAD

**Issue by KFaded22** [RPCS3/rpcs3#9740]
Currently UFC Undisputed 3 is in the best shape its been ever with RPCS3. 

Was actually able to create a fighter and start a career and finish a fight in exhibition once. 

Currently the issue with the game is that.. 

Once you get into a fight, it plays fine for about a minute into the 1st round of a fight, then the audio cuts out and everything 
goes silent - then about 20sh seconds after the audio cuts out the entire game freezes, sometimes it crashes RPCS3 all together. 

Also tried it out with the new [wip] texture cache rewrite build that was released today and it does the same thing. 

I've tried multiple combinations of settings and just about nothing changes it for the better. 

It also currently works better with ASMJIT than it does with LLVM. 

Multithreaded RSX is also not working in the game it makes it go from 55-60fps to only 1 to 2.33 fps. 

The game also goes up too 500%+ in speed in the games menus unless VSync is turned on. 

OS: Windows 10 

GPU: RX580 8gb 

RAM: 16gb 

Driver: 20.11.2

**Comment by Mookie19** [comment]
I'd like to add a few things from my testing

Multithreaded RSX does work, but doesn't seem to give any performance increase on my system, though it does seem to cause lots of graphical flickering, but certainly does not tank the FPS to 3.

SPU LLVM also seems to work better than SPU ASMJIT in regards to the length of time I can play (on average) until the game either freezes with the audio cut out or hard crashes. 

V-sync doesn't need to be on either, but a Framelimit of 60 is necessary (V-sync doesn't help me, since im only 144hz monitor)
If I limit the SPU threads to 1, I half my FPS essentially to a 30-40fps bracket instead of a steady 60 but the audio crackle is completely gone and play is elongated, but still audio will cut out and freeze eventually.

Driver-wake up delay doesn't seem to help with the hard crashes either (tried 0-5000, in 100us increments)

All my tests have been done with the same characters and in the same arena to eliminate any discrepancies aswell.

I can post logs anytime
Windows 10
Ryzen 5 3600X
RTX2070 - latest
32gb 3200mhz
RPCS3 0.0.15-11893.

**Comment by KFaded22** [comment]
> I'd like to add a few things from my testing
> 
> Multithreaded RSX does work, but doesn't seem to give any performance increase on my system, though it does seem to cause lots of graphical flickering, but certainly does not tank the FPS to 3.
> 
> SPU LLVM also seems to work better than SPU ASMJIT in regards to the length of time I can play (on average) until the game either freezes with the audio cut out or hard crashes.
> 
> V-sync doesn't need to be on either, but a Framelimit of 60 is necessary (V-sync doesn't help me, since im only 144hz monitor)
> If I limit the SPU threads to 1, I half my FPS essentially to a 30-40fps bracket instead of a steady 60 but the audio crackle is completely gone and play is elongated, but still audio will cut out and freeze eventually.
> 
> Driver-wake up delay doesn't seem to help with the hard crashes either (tried 0-5000, in 100us increments)
> 
> All my tests have been done with the same characters and in the same arena to eliminate any discrepancies aswell.
> 
> I can post logs anytime
> Windows 10
> Ryzen 5 3600X
> RTX2070 - latest
> 32gb 3200mhz
> RPCS3 0.0.15-11893.

That's really weird. For me Mulithreaded RSX just tanks the FPS to 3fps or worse. It only does this with the UFC games as well as far as my experience has been. 

LLVM definitely works better for the textures but ASMJIT would be more stable for me and longer play time. 

I guess it depends on the setup I suppose? 

I also only test in Career Mode - first fight in career and in Exhibition in the practice gym arena (less demanding) Always GSP and Matt Hughes as well

**Comment by Mookie19** [comment]
Whats your CPU, you never listed it? wondering if you're intel or not.

I do recall the Multithreaded RSX causing 3FPS while the SPU was set to 'fast' but I don't really contribute that to RSX threads.

Mark Hunt vs Mirko Cro Cop - Pride mode for my testing.

I read an issue here a few days ago about distortion effects causing spikes in GPU usage and I wonder if that has something to do with the hard crashes I get upon big 'rocked' moments due to it distorting the camera around the edges... who knows.

**Comment by KFaded22** [comment]
Ryzen 5 1600af (Zen 2 model) and that's about when it cuts out for me too when someone gets Rocked or near it happening. Always when I'm about to win a fight

**Comment by Mookie19** [comment]
Hey all, I have some updated info regarding BLES01231, IE UFC Undisputed 3 (UD3).

Firstly my system: Ryzen 3600X
                             32GB 3200mhz
                             2070 Super
                             RPCS3 on M.2 SSD

I'm not sure if this is system specific or not but that's why my specifications are there. 

The Audio cutting out - leading to a freeze momentarily later or an infinite loading screen (if it happened during a load) can be fixed (ON MY SYSTEM) with the debug option 'Hook Static Functions'. I'm unsure how this helps, but the description says 'either does nothing or breaks things'... Well it seems to actually fix the audio/freeze here for me.

The Second issue is the random crashing mid fight, sometimes randomly, other times when you land a big powerful hit. I've narrowed this specific crash down to also a debug setting, 'Force CPU Blit', which then requires write colour buffers. This setting is required to play the game with 'Proper' graphics/rendering.

The game is 90% playable with no (or rarely) any crashes and freezes with the only downside being improper rendering. Which in all honesty doesn't affect gameplay to, to much lol.

TLDR: Force CPU Blit causes the random crash in UFC Undisputed 3 and Hook Static Functions fixes the audio cut out/freeze (FOR ME). 

I'm unsure if a log will help with this issue or if you need a rsx capture but willing to provide anything needed!

**Comment by nustkyy** [comment]
Could you provide a screen shots or text with all your settings?

**Comment by nustkyy** [comment]
Is there anyway to fix graphics issues while having Force CPU Blit disabled?

**Comment by kjm5110** [comment]
> 안녕하세요, BLES01231, IE UFC Undisputed 3(UD3)에 대한 업데이트된 정보가 있습니다.
> 
> 먼저 내 시스템: M.2 SSD의 Ryzen 3600X
> 32GB 3200mhz
> 2070 Super
> RPCS3
> 
> 이것이 시스템 특정인지 아닌지 확실하지 않지만 이것이 제 사양이 있는 이유입니다.
> 
> 오디오 끊김 - 나중에 잠시 멈추거나 무한 로딩 화면(로드 중에 발생한 경우)을 디버그 옵션 'Hook Static Functions'로 고칠 수 있습니다(ON MY SYSTEM). 이것이 어떻게 도움이 되는지 확신할 수 없지만 설명에는 '아무것도 하지 않거나 문제가 발생합니다'라고 나와 있습니다... 글쎄, 실제로 여기에서 오디오/멈춤 현상을 수정하는 것 같습니다.
> 
> 두 번째 문제는 전투 중간에 무작위로 충돌하는 것입니다. 때로는 무작위로, 다른 때는 강력한 타격을 가할 때입니다. 필자는 이 특정 충돌을 디버그 설정인 'Force CPU Blit'으로 좁혔습니다. 그러면 쓰기 색상 버퍼가 필요합니다. 이 설정은 '적절한' 그래픽/렌더링으로 게임을 플레이하는 데 필요합니다.
> 
> 게임은 90% 플레이가 가능하며 충돌 및 정지가 전혀(또는 드물게) 발생하며 유일한 단점은 부적절한 렌더링입니다. 솔직히 말해서 게임 플레이에 영향을 미치지는 않습니다.
> 
> TLDR: Force CPU Blit은 UFC Undisputed 3에서 무작위 충돌을 일으키고 Hook Static Functions는 오디오 끊김/멈춤을 수정합니다(FOR ME).
> 
> 로그가 이 문제에 도움이 되는지 아니면 rsx 캡처가 필요하지만 필요한 것은 무엇이든 제공할 의향이 있는지 확실하지 않습니다!

Can you tell me the resolution settings.

**Comment by docemmettbrown1985** [comment]
@kjm5110 @KFaded22 @nustkyy @Mookie19 

"Is there a configuration method that does not cause conflict?"

Hi there!
I have reached this configuration that has helped me a lot to keep the game stable
I have not found one on the internet that works for me and I see that other users keep asking

I know that it works in INTEL, in my case it is AMD that I did not see much information either.


![CPU](https://user-images.githubusercontent.com/89216046/130119873-5b931cbf-3c3a-4a14-8d5f-dd8715675cd0.jpg)
![GPU](https://user-images.githubusercontent.com/89216046/130119877-15d1f2bd-504d-444c-a16a-a46b46760de7.jpg)
![ADVANCED](https://user-images.githubusercontent.com/89216046/130119881-c494673a-d34d-4132-be65-a88640e79445.jpg)

In Advanced Tab, the DRIVER WAKE-UP DELAY option, and GPU options, i tested and with this changes, works better, 
Without  doubts.

![AUDIO](https://user-images.githubusercontent.com/89216046/130119915-bb31d9a3-5ce1-4ff9-bac4-67b887ed3cfe.jpg)
![DEBUG](https://user-images.githubusercontent.com/89216046/130119918-3cefcda3-7cfe-4013-9110-a529d018cf27.jpg)

I couldn't figure out that the fighters don't look dark, attached a picture.

![In_game_little_dark](https://user-images.githubusercontent.com/89216046/130121990-2ce89898-57a5-4658-b216-d311ebd57276.jpg)

Please let me know if it worked for you


tested in 
- SYS: RPCS3 v0.0.17-12648-c13a46b0 Alpha | HEAD
- SYS: AMD Ryzen 7 2700 Eight-Core Processor | 16 Threads | 15.95 GiB RAM | TSC: 3.200GHz | AVX+ | FMA3
- SYS: Operating system: Windows, Major: 10
- RSX: Found vulkan-compatible GPU: 'Unknown AMD GPU' (RX580)

**Comment by rudz13** [comment]
Hi can any of you provide me with lic.dat trophy.dat and param sfo file of this game...it would be an great help ☺️

**Comment by Asinin3** [comment]
@rudz13 Buy the game and redump it, asking for game files here isn't allowed

**Comment by Megamouse** [comment]
this issue is pointless without a log
